File: backend/app/schemas/teacher.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime
from uuid import UUID

class TeacherBase(BaseModel):
    teacher_id: str
    full_name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    subject: Optional[str] = None
    status: str = 'active'

class TeacherCreate(TeacherBase):
    pass

class TeacherUpdate(BaseModel):
    teacher_id: Optional[str] = None
    full_name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    subject: Optional[str] = None
    status: Optional[str] = None

class TeacherInDB(TeacherBase):
    id: UUID
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/teacher.ts
// Teacher types aligned with backend schemas (app/schemas/teacher.py)

export interface Teacher {
  id: string;
  teacher_id: string;
  full_name: string;
  email: string | null;
  phone: string | null;
  subject: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface TeacherCreate {
  teacher_id: string;
  full_name: string;
  email?: string;
  phone?: string;
  subject?: string;
  status?: string;
}

export interface TeacherUpdate {
  teacher_id?: string;
  full_name?: string;
  email?: string;
  phone?: string;
  subject?: string;
  status?: string;
}
